dpo/aya_dpo_gen.py

```diff
@@ -0,0 +1,222 @@
+                resp = InferenceClient("laurievb/OpenLID").text_classification(
+                    cleaned_input
+                )
+                top_prediction = resp[
+                    0
+                ]  # top prediction is the first element in the list
+                input["predicted_generation_language"] = top_prediction.label
+                input["predicted_generation_language_score"] = top_prediction.score
+            except Exception as e:
+                print(e)
+                input["predicted_generation_language"] = "error"
+                input["predicted_generation_language_score"] = 0.0
+        yield inputs
+
+
+# Define a step to combine the Aya and model responses and add the response sources
+@step(inputs=["targets", "generation"], outputs=["generations"])
+def CombineAyaAndModelResponse(
+    inputs: StepInput,
+) -> StepOutput:
+    """A step to combine the Aya and model responses and add the response sources."""
+    for input in inputs:
+        input["generations"] = [input["targets"], input["generation"]]
+        input["response_source"] = ["aya", MODEL_ID]
+    yield inputs
+
+
+def update_argilla_dataset_with_metadata(
+    dataset_name: str,
+    workspace: str,
+    hub_dataset: Dataset,
+    metadata_keys: List[str] = None,
+):
+    """Update an Argilla dataset with metadata from the dataset."""
+    # by default, we add the predicted generation language and response source
+    if metadata_keys is None:
+        metadata_keys = ["predicted_generation_language", "response_source"]
+    dataset = rg.FeedbackDataset.from_argilla(
+        dataset_name,
+        workspace=workspace,
+    )
+
+    metadata_values = [hub_dataset[key] for key in metadata_keys]
+
+    if any(len(values) != len(dataset.records) for values in metadata_values):
+        raise ValueError(
+            f"Number of metadata values does not match the number of records ({len(dataset.records)})"
+        )
+
+    modified_records = []
+    for record, *metadata in zip(dataset.records, *metadata_values):
+        for key, value in zip(metadata_keys, metadata):
+            record.metadata[key] = value
+        modified_records.append(record)
+    dataset.update_records(modified_records)
+
+
+with Pipeline(name="generate-dpo-responses") as pipeline:
+    # Load the dataset from the Hugging Face Hub
+    load_hub_dataset = LoadHubDataset(
+        name="load_dataset",
+        output_mappings={"inputs": "instruction"},
+    )
+    # Generate responses using the model
+    text_generation = TextGeneration(
+        name="text_generation",
+        llm=InferenceEndpointsLLM(
+            base_url=INFERENCE_ENDPOINTS_URL,
+            tokenizer_id=MODEL_ID,
+            model_display_name=MODEL_ID,
+            api_key=HUGGINGFACE_TOKEN,
+        ),
+        input_batch_size=INPUT_BATCH_SIZE,
+        output_mappings={"model_name": "generation_model"},
+        num_generations=2,
+    )
```

Comment:
If you find that the quality of responses is low, or included non dutch prose, you might want to try a custom prompt by using a custom TextGeneration step: https://distilabel.argilla.io/latest/sections/learn/tasks/text_generation/#using-a-custom-prompt